include/dali/core/traits.h

```diff
@@ -42,6 +44,32 @@ using remove_cv_t = std::remove_cv_t<T>;
 template <bool Value, typename Type = void>
 using enable_if_t = std::enable_if_t<Value, Type>;
 
+
+template <typename T, typename _ = void>
+struct is_container : std::false_type {};
+
+template <typename... Ts>
+struct is_container_helper {};
+
+template <typename T>
+struct is_container<T,
```

Comment:
It may be a nice trait on its own, but not really applicable for detecting span-compatible containers - we need more than iterators for that.